### PR TITLE
Add emacs mode

### PR DESCRIPTION
**Pull request by azzamsa**

### CODE REVIEW

File: contrib/emacs/hurl-mode.el [hunk 1]
```diff
@@ -0,0 +1,136 @@
+;;; hurl-mode.el --- Major mode for hurl  -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2022 Azzam S.A
```
Comment (azzamsa) [inline contrib/emacs/hurl-mode.el:3]: @jcamiel this need to be changed. For reference:
- https://github.com/spotify/dockerfile-mode/blob/52c6c00da1d31c0b6c29c74335b3af63ed6bf06c/dockerfile-mode.el#L3

File: contrib/emacs/hurl-mode.el [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+;; Author: Azzam S.A <vcs@azzamsa.com>
```
Comment (azzamsa) [inline contrib/emacs/hurl-mode.el:5]: We can add more names here such this:
- https://github.com/magit/magit/blob/99d9b3008adf72a0c2cdb7df70b5ae03ea9271c8/lisp/magit.el#L5

Comment (jcamiel) [reply contrib/emacs/hurl-mode.el:5]: Same, here, could you please remove:

`;; Copyright (C) 2022 Azzam S.A`
`;; Author: Azzam S.A <vcs@azzamsa.com>` 

(you contribution wil be of course attributed to you, you will be able to check it with git)

File: contrib/emacs/hurl-mode.el [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+;; Homepage: https://github.com/Orange-OpenSource/hurl
+;; Keywords: Hurl, shell
+
+;; Package-Version: 0.1.0
+;; Package-Requires: ((emacs "24"))
+
+;; SPDX-License-Identifier: GPL-3.0-or-later
```
Comment (jcamiel) [inline contrib/emacs/hurl-mode.el:12]: Change this to 'Apache License 2.0` and remove the body of the license. We have a [LICENSE](https://github.com/Orange-OpenSource/hurl/blob/master/LICENSE) file that is the unique reference of the whole project.

**Comment by jcamiel** [comment]
Just another thing, our [CONTRIBUTING.md](https://github.com/Orange-OpenSource/hurl/blob/master/CONTRIBUTING.md) require to be signed and marked as "Verified" (signed with a GPG, SSH, or S/MIME that is successfully verified by GitHub). You can see check how to do this here <https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification>. I can assist you if you need!

File: contrib/emacs/hurl-mode.el [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+;; Homepage: https://github.com/Orange-OpenSource/hurl
+;; Keywords: Hurl, shell
+
+;; Package-Version: 0.1.0
+;; Package-Requires: ((emacs "24"))
+
+;; SPDX-License-Identifier: Apache License 2.0
+
+;; This program is free software; you can redistribute it and/or modify
```
Comment (jcamiel) [inline contrib/emacs/hurl-mode.el:11]: Sorry @azzamsa everything looks good except this: can you remove line 11 to 22 (everything related to license). We should just set `SPDX-License-Identifier: Apache License 2.0` and nothing else as it is sufficient to make our license explicit. Thanks again!!

Comment (azzamsa) [reply contrib/emacs/hurl-mode.el]: Oh sorry. I missed this.

**Comment by jcamiel** [comment]
@azzamsa Could you please rebase your PR to appease our CI ? I've try to update your fork but I didn't have the right to push on it, thanks!

**Comment by azzamsa** [comment]
@jcamiel done.

**Comment by azzamsa** [comment]
@jcamiel one test is failed. Is there anything I can do about it?

- https://github.com/Orange-OpenSource/hurl/actions/runs/3716735257/jobs/6303402352#step:13:30

Looks like outdated dependencies.

**Comment by jcamiel** [comment]
@azzamsa I will to the updates. In the meantime would you be possible to put your PR on a branch (rather on master)? I think this is why I can't edit your fork (unless you've activated the option to not allow 'edit by maintainers'). This way, as your PR is OK, I can manage everything instead of asking you to rebase your work (currently I can't edit your fork, I don't know why). We can try this!

**Pull request closed without merging**